users/app.py

```diff
@@ -54,15 +54,47 @@ def add_update_user():
     return jsonify(user_object), (201 if added_new_user else 200)
 
 
+def upsert_user_in_db(user_object, users_collection):
+    """Updates or inserts the user object into user_collection.
+
+    Adds the is_oganizer field to user_object with default value False.
+
+    user_object should look like
+        {"user_id": foobar,
+         "name": Foo Bar}
+
+    Documents in users_collection should look like
+        {"user_id": foobar,
+         "name": Foo Bar,
+         "is_organizer": False}
+
+    Returns:
+        bool: whether user_object was formatted correctly and the upsert
+            was successful.
+    """
+    if "user_id" not in user_object or "name" not in user_object:
```

Comment:
Consider using the dict view returned by `.keys()` to compare with a set of fields:

```python
if user_object.keys() != set("user_id", "name"):
  return False  # Why not throw?
```

Reply:
Reading your tests below, it sounds like you would want something like:

```python
if user_object.keys() <= set("user_id", "name"):
```

Reply:
Works with
```python
user_object.keys() < {"user_id", "name"}
```
Fixed in https://github.com/knative-portability/large-events/pull/8/commits/25039520f28fc4f4ce9d5194d2004f2499ab2ef6